### ISSUE TITLE
Not able to redirect to Angular url using #

### ISSUE THREAD

**Issue by simonbernard** [TryGhost/Ghost#10290]
### Issue Summary

Using redirects.json I try to redirect to `https://someurl.com/#/login`. Ghost encodes the URL to `https://someurl.com/%23/login` so it is not recognized by Angular anymore.

### To Reproduce

1. Upload this redirects.json:

```json
[
  {
    "from": "/login/" ,
    "to":  "https://someurl.com/#/login",
    "permanent": true
  }
]
```

2. Test the redirected URL, it forwards to encoded URL `https://someurl.com/%23/login`

I understand it can be useful or even neccessarry to encode the URL, but # is a reserved character so it is not neccessarry to encode it - maybe you can let it up to the user to enter encoded URL directly?

### Technical details:

* Ghost Version: Using docker image 2.6.2-alpine
* Node Version: 8.14.0
* Browser/OS: Current Firefox on MacOS
* Database: sqlite3

**Thanks very much for your great work!**

**Comment by kirrg001** [comment]
Hey @simonbernard!

Thanks for your report.

`/#/login` is a hash. Ghost currently uses `url.format` which assumes that `to` is a path. 
`url.format` [auto encodes](https://github.com/nodejs/node-v0.x-archive/issues/4082) path delimiters like `#` or `?`.

To support your use case, i think we should detect `^/#/...` and forward it as `hash` option to `url.format`.

PR is welcome 👍

**Comment by greghart** [comment]
I was going to tackle this thinking it'd be simple, but now I've gone and gotten confused!

I think the issue with special casing `#` (or `?`, it's a similar issue), is how one should coalesce hash or search params in the incoming URL with hash or search params in the `to` property of `redirects.json`.

## Current Scenario
Eg. as it works right now, with no hash or search params in the `to`
```json 
{ "from": "/foo", "to": "/bar" }
```
redirects happen as:
```
1: /foo -> /bar
2: /foo?search=test -> /bar?search=test
3: /foo#hash -> /bar#hash
4: /foo?search=test#hash -> /bar?search=test#hash
```

So the incoming hash/search is propagated through as well. 

## Given Scenario
In the given scenario of including hash or search params in the `to`
```json
{ "from": "/foo", "to": "/bar?search=other#login" }
```
I'm not sure what the "right" way would be for use cases 2, 3, and 4, and whether we need to keep backwards compatibility with the current scenario.
```
1: /foo -> /bar?search=test#hash
2: /foo?search=test -> ???
3: /foo#hash -> ???
4: /foo?search=test#hash -> ???
```

tl;dr: we potentially have two sources of hash/search data, and it's not obvious how to handle that.

**Comment by greghart** [comment]
That's a good point, so really it's the browser that decides how hashes are coalesced. 

On Chrome at least, the browser will propagate an existing hash through a redirect *unless* a hash is given by the server, in which case browser will apply that.

For example:
User navigates to `/foo1#hash`, gets a redirect to `/bar1`, and browser keeps the hash, so user sees `/bar1#hash`.
User navigates to `/foo2#hash`, gets a redirect to `/bar2#login`, and browser uses that, so user sees `/bar2#login`. 

Really, I only brought it up as I was thinking of how search params would work, but that may be getting ahead of this use case. For now I'll keep things simple and implement kirgg's proposed solution, and if people come up with a use case for dropping search params in `to` also we can deal with that then.
